python/cudf/cudf/options.py

```diff
@@ -114,6 +115,40 @@ def set_option(name: str, val: Any):
     option.value = val
 
 
+class option_context(ContextDecorator):
+    """
+    Context manager to temporarily set options in the `with` statement context.
+
+    You need to invoke as ``option_context(pat, val, [(pat, val), ...])``.
+
+    Examples
+    --------
+    >>> from cudf import option_context
+    >>> with option_context('display.max_rows', 10, 'display.max_columns', 5):
```

Comment:
I tried to read the API docs and answer this question but I couldn't. Does a cudf option for `display.max_rows` exist? Do the options propagate from cudf to pandas somehow? I know pandas' options for `display.max_rows` will affect cudf, but wasn't sure about the opposite. If this doesn't exist, we should change the docstring example to something that exists.

Also we need to explicitly list all cudf options someplace in our docs. I didn't find it when I looked. Does such a list exist?

Reply:
> I tried to read the API docs and answer this question but I couldn't. Does a cudf option for display.max_rows exist? Do the options propagate from cudf to pandas somehow? I know pandas' options for display.max_rows will affect cudf, but wasn't sure about the opposite. If this doesn't exist, we should change the docstring example to something that exists.

Given that we rely on pandas, should we migrate the cudf options to just _using_ the pandas options object, and namespace any cudf-specific options behind `cudf.WHATEVER`. That way we delete code (good), and don't have to worry about keeping things in sync.

> Also we need to explicitly list all cudf options someplace in our docs. I didn't find it when I looked. Does such a list exist?

I think no.

Reply:
@wence- I think we decided against reusing pandas' options directly because we want to ensure that cudf has control of its own options if pandas decides to change its API, or place restrictions on what constitutes a valid key or value. There is also an implicit expectation that if cudf uses pandas' options, it also _supports_ all pandas options -- which we do not.

Reply:
Thanks, makes sense.

Reply:
I updated the docstring with cudf specific parameters. Also, added the available options section to our docs.